src/looprestoration.rs

```diff
@@ -23,3 +27,188 @@ pub union LooprestorationParams {
     pub filter: Align16<[[int16_t; 8]; 2]>,
     pub sgr: LooprestorationParams_sgr,
 }
+
+macro_rules! looprestoration_filter_fn_enum {
```

Comment:
This is a cool macro.  I didn't think of making them this long, especially in the pattern/match part of it.